src/mountscript.sh

```diff
@@ -741,243 +167,60 @@ parse_arguments()
-# error.
-#
-ensure_aznfswatchdog()
-{
-    pidof -x aznfswatchdog > /dev/null 2>&1
-    if [ $? -ne 0 ]; then
-        if systemd_is_init; then
-            eecho "aznfswatchdog service not running!"
-            pecho "Start the aznfswatchdog service using 'systemctl start aznfswatchdog' and try again."
-        else
-            eecho "aznfswatchdog service not running, please make sure it's running and try again!"
-        fi
-        
-        pecho "If the problem persists, contact Microsoft support."
-        return 1
-    fi
-}
-
-#
-# To maintain consistency in case of regional account and in general to avoid creating
-# multiple DNAT entries corrosponding to one LOCAL_IP, first check for resolved IP in mountmap.
-# This will help keep mountmap and DNAT entries in sync with each other.
-# If the current resolved IP is different from the one stored in mountmap then it means that the IP has changed
-# since the mountmap entry was created (could be due to migration or more likely due to RAs roundrobin DNS). 
-# In any case this will be properly handled by aznfswatchdog next time it checks for IP change for this fqdn.
-#
-resolve_ipv4_with_preference_to_mountmap()
-{
-    local fqdn=$1
-
-    exec {fd}<$MOUNTMAP
-    flock -e $fd
-
-    local mountmap_entry=$(grep -m1 "^${fqdn} " $MOUNTMAP)
-    
-    flock -u $fd
-    exec {fd}<&-
+# [account.blob.core.windows.net:/account/container /mnt/aznfs -o rw,tcp,nolock,nconnect=16]
+vecho "Got arguments: [$*]"
 
-    IFS=" " read _ local_ip old_nfs_ip <<< "$mountmap_entry"
-    if [ -n "$old_nfs_ip" ]; then
-        echo "$old_nfs_ip"
-        return 2 
-    fi
+mount_point="$2"
 
-    #
-    # Resolve FQDN to IPv4 using DNS if not found in the mountmap.
-    #
-    resolve_ipv4 "$fqdn" "true"
-}
+OPTIONS=
+MOUNT_OPTIONS=
+AZ_PREFIX=
 
-# [account.blob.core.windows.net:/account/container /mnt/aznfs -o rw,tcp,nolock,nconnect=16]
-vecho "Got arguments: [$*]"
+parse_arguments $*
 
-# Check if aznfswatchdog service is running.
-if ! ensure_aznfswatchdog; then
+nfs_vers=$(get_version_from_mount_options "$MOUNT_OPTIONS")
+if [ $? -ne 0 ]; then
+    echo "$nfs_vers"
     exit 1
 fi
 
-# MOUNTMAP file must have been created by aznfswatchdog service.
-if [ ! -f "$MOUNTMAP" ]; then
-    eecho "[FATAL] ${MOUNTMAP} not found!"
-    
-    if systemd_is_init; then
-        pecho "Try restarting the aznfswatchdog service using 'systemctl start aznfswatchdog' and then retry the mount command."
-    else
-        eecho "aznfswatchdog service not running, please make sure it's running and try again!"
-    fi
-    
-    pecho "If the problem persists, contact Microsoft support."
+if [ $nfs_vers == "4.1" ]; then
+    AZ_PREFIX="file"
+elif [ $nfs_vers == "3" ]; then
+    AZ_PREFIX="blob"
+else
+    eecho "NFS version is not supported by mount helper: $nfs_vers!"
     exit 1
 fi
 
-nfs_host=$(get_host_from_share "$1")
+nfs_host=$(get_host_from_share "$1" "$AZ_PREFIX")
```

Comment:
@linuxsmiths Since $AZ_PREFIX is global var, is it needs to be passed in these functions?